include/nbl/builtin/hlsl/spirv_intrinsics/glsl.std.450.hlsl

```diff
@@ -0,0 +1,86 @@
+#include <nbl/builtin/hlsl/concepts.hlsl>
+#include "spirv/unified1/GLSL.std.450.h"
+
+namespace nbl 
+{
+namespace hlsl
+{
+namespace spirv
+{
+// Find MSB and LSB restricted to 32-bit width component types https://registry.khronos.org/SPIR-V/specs/unified1/GLSL.std.450.html
+template<typename Integral32 NBL_FUNC_REQUIRES(is_same_v<Integral32, int32_t> || is_same_v<Integral32, uint32_t>)
+[[vk::ext_instruction(GLSLstd450::GLSLstd450FindILsb, "GLSL.std.450")]]
+Integral32 findILsb(Integral32 value);
+
+template<int N>
+[[vk::ext_instruction(GLSLstd450::GLSLstd450FindILsb, "GLSL.std.450")]]
+vector<int32_t, N> findILsb(vector<int32_t, N> value);
+
+template<int N>
+[[vk::ext_instruction(GLSLstd450::GLSLstd450FindILsb, "GLSL.std.450")]]
+vector<uint32_t, N> findILsb(vector<uint32_t, N> value);
+
+template<typename Int32_t NBL_FUNC_REQUIRES(is_same_v<Int32_t, int32_t>)
+[[vk::ext_instruction(GLSLstd450::GLSLstd450FindSMsb, "GLSL.std.450")]]
+int32_t findSMsb(Int32_t value);
+
+template<int N>
+[[vk::ext_instruction(GLSLstd450::GLSLstd450FindSMsb, "GLSL.std.450")]]
+vector<int32_t, N> findSMsb(vector<int32_t, N> value);
+
+template<typename Uint32_t NBL_FUNC_REQUIRES(is_same_v<Uint32_t, uint32_t>)
+[[vk::ext_instruction(GLSLstd450::GLSLstd450FindUMsb, "GLSL.std.450")]]
+int32_t findUMsb(Uint32_t value);
+
+template<int N>
+[[vk::ext_instruction(GLSLstd450::GLSLstd450FindUMsb, "GLSL.std.450")]]
+vector<uint32_t, N> findUMsb(vector<uint32_t, N> value);
+
+template<typename FloatingPoint>
+[[vk::ext_instruction(GLSLstd450::GLSLstd450Exp2, "GLSL.std.450")]]
+enable_if_t<is_floating_point<FloatingPoint>::value && !is_matrix_v<FloatingPoint>, FloatingPoint> exp2(FloatingPoint val);
+
+template<typename FloatingPoint>
+[[vk::ext_instruction(GLSLstd450::GLSLstd450InverseSqrt, "GLSL.std.450")]]
+enable_if_t<is_floating_point_v<FloatingPoint> && !is_matrix_v<FloatingPoint>, FloatingPoint> inverseSqrt(FloatingPoint val);
+ 
+template<typename FloatingPoint>
+[[vk::ext_instruction(GLSLstd450::GLSLstd450Floor, "GLSL.std.450")]]
+enable_if_t<is_floating_point_v<FloatingPoint> && !is_matrix_v<FloatingPoint>, FloatingPoint> floor(FloatingPoint val);
+
+template<typename FloatingPoint>
+[[vk::ext_instruction(GLSLstd450::GLSLstd450Cross, "GLSL.std.450")]]
+enable_if_t<is_floating_point_v<FloatingPoint>, vector<FloatingPoint, 3> > cross(in vector<FloatingPoint, 3> lhs, in vector<FloatingPoint, 3> rhs);
+
+template<typename FloatingPoint>
+[[vk::ext_instruction(GLSLstd450::GLSLstd450FMix, "GLSL.std.450")]]
+enable_if_t<is_floating_point_v<FloatingPoint> && !is_matrix_v<FloatingPoint>, FloatingPoint> fMix(FloatingPoint x, FloatingPoint y, FloatingPoint a);
+
+template<typename T, int N>
+[[vk::ext_instruction(GLSLstd450::GLSLstd450Determinant, "GLSL.std.450")]]
+T determinant(in matrix<T, N, N> mat);
+
+template<typename T, int N>
+[[vk::ext_instruction(GLSLstd450MatrixInverse, "GLSL.std.450")]]
+matrix<T, N, N> matrixInverse(in matrix<T, N, N> mat);
```

Comment:
why are you using `in`  keyword on some parameters in the signatures? (also `cross`)

Reply:
Also wondering why GH diff shows me this file even though its identical as the one in `master`

Reply:
> why are you using `in` keyword on some parameters in the signatures? (also `cross`)

changed to `NBL_CONST_REF_ARG`

Reply:
for spirv intrinsics you shouldn't need to use `in`, `out` or anything like that.

If you truly need a reference (SPIR-V spec says its a pointer) then use `vk::ext_reference` but that stuff is only for stuff that NEEDs a reference (atomics) or cannot be copied (ray query)